File: fsharp/say/SayTest.fs
module SayTest

open NUnit.Framework

open Say

[<Test>]
let ``Zero`` () =
    Assert.That(inEnglish 0L, Is.EqualTo(Some "zero"))

[<Test>]
let ``One`` () =
    Assert.That(inEnglish 1L, Is.EqualTo(Some "one"))

[<Test>]
let ``Fourteen`` () =
    Assert.That(inEnglish 14L, Is.EqualTo(Some "fourteen"))

[<Test>]
let ``Twenty`` () =
    Assert.That(inEnglish 20L, Is.EqualTo(Some "twenty"))

[<Test>]
let ``Twenty-two`` () =
    Assert.That(inEnglish 22L, Is.EqualTo(Some "twenty-two"))

[<Test>]
let ``One hundred`` () =
    Assert.That(inEnglish 100L, Is.EqualTo(Some "one hundred"))

[<Test>]
let ``One hundred twenty-three`` () =
    Assert.That(inEnglish 123L, Is.EqualTo(Some "one hundred twenty-three"))

[<Test>]
let ``One thousand`` () =
    Assert.That(inEnglish 1000L, Is.EqualTo(Some "one thousand"))

[<Test>]
let ``One thousand two hundred thirty-four`` () =
    Assert.That(inEnglish 1234L, Is.EqualTo(Some "one thousand two hundred thirty-four"))

[<Test>]
let ``One million`` () =
    Assert.That(inEnglish 1000000L, Is.EqualTo(Some "one million"))

[<Test>]
let ``One million two`` () =
    Assert.That(inEnglish 1000002L, Is.EqualTo(Some "one million two"))

[<Test>]
let ``One million two thousand three hundred forty-five`` () =
    Assert.That(inEnglish 1002345L, Is.EqualTo(Some "one million two thousand three hundred forty-five"))

[<Test>]
let ``One billion`` () =
    Assert.That(inEnglish 1000000000L, Is.EqualTo(Some "one billion"))

[<Test>]
let ``A big number`` () =
    Assert.That(inEnglish 987654321123L, Is.EqualTo(Some "nine hundred eighty-seven billion six hundred fifty-four million three hundred twenty-one thousand one hundred twenty-three"))
 
[<Test>]
let ``Lower bound`` () =
    Assert.That(inEnglish -1L, Is.EqualTo(None))

[<Test>]
let ``Upper bound`` () =
    Assert.That(inEnglish 1000000000000L, Is.EqualTo(None))

File: csharp/say/SayTest.cs
using NUnit.Framework;

public class SayTest
{
    [Test]
    public void Zero()
    {
        Assert.That(Say.InEnglish(0L), Is.EqualTo("zero"));
    }

    [Test]
    public void One()
    {
        Assert.That(Say.InEnglish(1L), Is.EqualTo("one"));
    }

    [Test]
    public void Fourteen()
    {
        Assert.That(Say.InEnglish(14L), Is.EqualTo("fourteen"));
    }

    [Test]
    public void Twenty()
    {
        Assert.That(Say.InEnglish(20L), Is.EqualTo("twenty"));
    }

    [Test]
    public void Twenty_two()
    {
        Assert.That(Say.InEnglish(22L), Is.EqualTo("twenty-two"));
    }

    [Test]
    public void One_hundred()
    {
        Assert.That(Say.InEnglish(100L), Is.EqualTo("one hundred"));
    }

    [Test]
    public void One_hundred_twenty_three()
    {
        Assert.That(Say.InEnglish(123L), Is.EqualTo("one hundred twenty-three"));
    }

    [Test]
    public void One_thousand()
    {
        Assert.That(Say.InEnglish(1000L), Is.EqualTo("one thousand"));
    }

    [Test]
    public void One_thousand_two_hundred_thirty_four()
    {
        Assert.That(Say.InEnglish(1234L), Is.EqualTo("one thousand two hundred thirty-four"));
    }

    [Test]
    public void One_million()
    {
        Assert.That(Say.InEnglish(1000000L), Is.EqualTo("one million"));
    }

    [Test]
    public void One_million_two()
    {
        Assert.That(Say.InEnglish(1000002L), Is.EqualTo("one million two"));
    }

    [Test]
    public void One_million_two_thousand_three_hundred_forty_five()
    {
        Assert.That(Say.InEnglish(1002345L), Is.EqualTo("one million two thousand three hundred forty-five"));
    }

    [Test]
    public void One_billion()
    {
        Assert.That(Say.InEnglish(1000000000L), Is.EqualTo("one billion"));
    }

    [Test]
    public void A_big_number()
    {
        Assert.That(Say.InEnglish(987654321123L), Is.EqualTo("nine hundred eighty-seven billion six hundred fifty-four million three hundred twenty-one thousand one hundred twenty-three"));
    }

    [Test]
    public void Lower_bound()
    {
        Assert.That(() => Say.InEnglish(-1L), Throws.Exception);
    }

    [Test]
    public void Upper_bound()
    {
        Assert.That(() => Say.InEnglish(1000000000000L), Throws.Exception);
    }
}